src/utils/useActivation.ts

```diff
@@ -19,75 +20,98 @@ interface ActivationResult {
   const isAvailable = useIsHeadlessAvailable();
-  const [isPolling, setPolling] = useState(false);
+  const [activateAccountTxId, setActivateAccountTxId] = useState<string>();
   const [txError, setTxError] = useState<Error | undefined>();
   const { loading, data, error } = useActivationAddressQuery({
     variables: {
-      address: account.address,
+      address,
     },
-    pollInterval: isPolling ? 1000 : undefined,
-    skip: !account.isLogin,
+    pollInterval: activateAccountTxId ? 1000 : undefined,
+    skip: !address,
   });
+  const tx = useTx();
+  const activated = data?.activationStatus.addressActivated ?? false;
 
   const {
     loading: nonceLoading,
     data: nonceData,
     error: nonceError,
   } = useActivationKeyNonceQuery({
     variables: {
-      // @ts-expect-error The query will not run if activationKey is undefined due to the skip option.
-      encodedActivationKey: activationKey,
+      encodedActivationKey: accountStore.activationKey,
+    },
+    skip: !accountStore.activationKey,
+  });
+  const [
+    requestActivateAccountTx,
+    {
+      called: activateAccountTxCalled,
+      loading: activateAccountTxLoading,
+      error: activateAccountTxError,
+    },
+  ] = useActivateAccountLazyQuery({
+    fetchPolicy: "network-only",
+    onCompleted: ({ actionTxQuery: { activateAccount } }) => {
+      tx(activateAccount)
+        .then((res) => {
+          if (res.data?.stageTransaction) {
+            setActivateAccountTxId(res.data.stageTransaction);
+          }
+        })
+        .catch((e) => {
+          setTxError(e);
+        });
     },
-    skip: !activationKey,
   });
-  const [tx] = useActivateAccountLazyQuery();
 
   useEffect(() => {
     if (nonceData?.activated) {
       setTxError(new Error("Already activated."));
       return;
     }
+
+    if (!doActivate) {
+      return;
+    }
+
     if (
-      !data?.activationStatus.addressActivated &&
-      activationKey &&
-      nonceData?.activationKeyNonce &&
-      !isPolling
+      !activated &&
+      !activateAccountTxCalled &&
+      !activateAccountTxLoading &&
+      publicKey &&
+      accountStore.activationKey &&
+      nonceData?.activationKeyNonce
     ) {
-      unstable_batchedUpdates(() => {
-        setPolling(true);
-        setTxError(undefined);
+      requestActivateAccountTx({
+        variables: {
+          publicKey,
+          activationCode: accountStore.activationKey,
+        },
       });
-      account
-        .getPublicKeyString()
-        .then((v) =>
-          tx({
-            variables: {
-              publicKey: v,
-              activationCode: activationKey,
-            },
-          })
-        )
-        .catch((e) => {
-          setTxError(e);
-          console.error(e);
-        })
-        .then(() => setPolling(false));
     }
-  }, [activationKey, tx, nonceData, isPolling]);
-
-  useEffect(() => {
-    if (data?.activationStatus.addressActivated) setPolling(false);
-  }, [data]);
+  }, [
```

Comment:
```suggestion
 if(activated) stopPolling();
   }, [
```

We should stop polling after activated.